### PR TITLE
wip, pending investigation on hardcoding values.

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

closes: #XXXX

### Commit Message / Changelog Entry

```bash
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/04-channel/types/msgs_test.go [hunk 1]
```diff
@@ -350,14 +351,14 @@ func (suite *TypesTestSuite) TestMsgChannelCloseConfirmValidateBasic() {
...
 	}{
-		{"", types.NewMsgChannelCloseConfirm(portid, chanid, suite.proof, height, addr), true},
-		{"too short port id", types.NewMsgChannelCloseConfirm(invalidShortPort, chanid, suite.proof, height, addr), false},
-		{"too long port id", types.NewMsgChannelCloseConfirm(invalidLongPort, chanid, suite.proof, height, addr), false},
-		{"port id contains non-alpha", types.NewMsgChannelCloseConfirm(invalidPort, chanid, suite.proof, height, addr), false},
-		{"too short channel id", types.NewMsgChannelCloseConfirm(portid, invalidShortChannel, suite.proof, height, addr), false},
-		{"too long channel id", types.NewMsgChannelCloseConfirm(portid, invalidLongChannel, suite.proof, height, addr), false},
-		{"channel id contains non-alpha", types.NewMsgChannelCloseConfirm(portid, invalidChannel, suite.proof, height, addr), false},
-		{"empty proof", types.NewMsgChannelCloseConfirm(portid, chanid, emptyProof, height, addr), false},
+		{"", types.NewMsgChannelCloseConfirm(portid, chanid, suite.proof, height, addr, 0), true},
```
Comment (colin-axner) [inline modules/core/04-channel/types/msgs_test.go:354]: if there's a constant for the counteparty upgrade sequence, you could use that instead of 0, but I'm perfectly fine with this. Feel free to resolve

File: modules/core/04-channel/types/msgs_test.go [hunk 2]
```diff
@@ -350,14 +351,14 @@ func (suite *TypesTestSuite) TestMsgChannelCloseConfirmValidateBasic() {
 		msg     *types.MsgChannelCloseConfirm
 		expPass bool
 	}{
```
Comment (colin-axner) [inline modules/core/04-channel/types/msgs_test.go:353]: super super nit, but should we add a success test case for non zero counterparty upgrade sequence?

File: modules/core/04-channel/types/msgs_test.go [hunk 3]
```diff
@@ -472,12 +473,12 @@ func (suite *TypesTestSuite) TestMsgTimeoutOnCloseValidateBasic() {
 		msg     *types.MsgTimeoutOnClose
 		expPass bool
 	}{
-		{"success", types.NewMsgTimeoutOnClose(packet, 1, suite.proof, suite.proof, height, addr), true},
-		{"seq 0", types.NewMsgTimeoutOnClose(packet, 0, suite.proof, suite.proof, height, addr), false},
-		{"signer address is empty", types.NewMsgTimeoutOnClose(packet, 1, suite.proof, suite.proof, height, emptyAddr), false},
-		{"empty proof", types.NewMsgTimeoutOnClose(packet, 1, emptyProof, suite.proof, height, addr), false},
-		{"empty proof close", types.NewMsgTimeoutOnClose(packet, 1, suite.proof, emptyProof, height, addr), false},
-		{"invalid packet", types.NewMsgTimeoutOnClose(invalidPacket, 1, suite.proof, suite.proof, height, addr), false},
+		{"success", types.NewMsgTimeoutOnClose(packet, 1, suite.proof, suite.proof, height, addr, 0), true},
```
Comment (colin-axner) [inline modules/core/04-channel/types/msgs_test.go:476]: ditto for non-zero counterparty upgrade sequence

File: modules/core/04-channel/keeper/timeout_test.go [hunk 4]
```diff
@@ -718,8 +719,9 @@ func (suite *KeeperTestSuite) TestTimeoutOnClose() {
 		suite.Run(fmt.Sprintf("Case %s, %d/%d tests", tc.msg, i, len(testCases)), func() {
 			var proof []byte
 
-			suite.SetupTest() // reset
-			nextSeqRecv = 1   // must be explicitly changed
+			suite.SetupTest()               // reset
+			nextSeqRecv = 1                 // must be explicitly changed
+			counterpartyUpgradeSequence = 0 // must be explicitly changed
```
Comment (colin-axner) [inline modules/core/04-channel/keeper/timeout_test.go:724]: should this have a test case for a invalid counterparty upgrade sequence?

File: modules/core/04-channel/keeper/handshake_test.go [hunk 5]
```diff
@@ -799,8 +800,9 @@ func (suite *KeeperTestSuite) TestChanCloseConfirm() {
 	for _, tc := range testCases {
 		tc := tc
 		suite.Run(fmt.Sprintf("Case %s", tc.msg), func() {
-			suite.SetupTest() // reset
-			heightDiff = 0    // must explicitly be changed
+			suite.SetupTest()               // reset
+			heightDiff = 0                  // must explicitly be changed
+			counterpartyUpgradeSequence = 0 // must explicitly be changed
```
Comment (colin-axner) [inline modules/core/04-channel/keeper/handshake_test.go:805]: add invalid counterparty upgrade sequence test case?

**Review by colin-axner — APPROVED** [review]
Nice work! Left a few suggestions on the testing front

**Review by colin-axner — APPROVED** [review]
ACK latest changes, thanks!

**Review by damiannolan — APPROVED** [review]
ACK, great work @DimitrisJim! Thanks ❤️ 

LGTM!!!
